src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -189,8 +189,7 @@ public Iterator<Path> iterator() {
                         sm.checkRead(rootDirectory.toString());
                     return allowedList.iterator();
                 } catch (SecurityException x) {
-                    List<Path> disallowed = Collections.emptyList();
-                    return disallowed.iterator();
+                    return Collections.emptyIterator(); //disallowed
```

Comment:
Alternatively, you could use the exception parameter to capture that comment:

    } catch (SecurityException disallowed) {